### PR TITLE
[CALCITE-6140] Add percentile_rank test on SqlParserTest

### PR DESCRIPTION
**Pull request by caicancai**
https://issues.apache.org/jira/browse/CALCITE-6140

### CODE REVIEW

File: testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java [hunk 1]
```diff
@@ -784,6 +784,13 @@ protected static SortedSet<String> keywords(@Nullable String dialect) {
         .fails("(?s)Encountered \"\\*\" at .*");
   }
 
+
+  @Test void testPercentile() {
+    sql("SELECT percentile(x, .5) within group (order by 3) from t")
+        .ok("SELECT `PERCENTILE`(`X`, 0.5) WITHIN GROUP (ORDER BY 3)\n"
```
[the remark this answers is not in the document]
Comment (caicancai) [reply testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java:790]: It seems that there is no such function in many databases, I only know that hive supports this function, but I see that sqlParserTest is marked with this function, should we delete this mark

**Comment by caicancai** [comment]
Screenshot_12

File: testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tanclary) [reply testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java:790]: If hive offers a `PERCENTILE` function, I don't think Calcite supports it yet. In that case, this test should probably wait until (at least) parsing is available.

[the remark this answers is not in the document]
Comment (caicancai) [reply testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java]: Thank you for your review and reply. I changed test to PERCENT_RANK. pg and sqlserver support it.

**Pull request closed without merging**